app/invocation/invocation_action_card.tsx

```diff
@@ -0,0 +1,165 @@
+import pako from "pako";
+import React from "react";
+import format from "../format/format";
+import InvocationModel from "./invocation_model";
+import { build } from "../../proto/remote_execution_ts_proto";
+import rpcService from "../service/rpc_service";
+
+interface Props {
+  model: InvocationModel;
+  search: string;
+}
+
+interface State {
+  contents?: ArrayBuffer;
+  action_array?: Uint8Array;
+  action?: build.bazel.remote.execution.v2.Action;
+  command?: build.bazel.remote.execution.v2.Command;
+  error?: string;
+}
+
+export default class ActionCardComponent extends React.Component<Props, State> {
+  state: State = {};
+  componentDidMount() {
+    this.fetchAction();
+  }
+
+  fetchAction() {
+    // TODO: Replace localhost:1987 with the remote cache address and prefix the path with the instance name
+    let actionFile = "bytestream://localhost:1987/blobs/" + this.props.search.substring(14);
+    rpcService
+      .fetchBytestreamFile(actionFile, this.props.model.getId(), "arraybuffer")
+      .then((action_buff: any) => {
+        let temp_array = new Uint8Array(action_buff);
+        let temp_action = build.bazel.remote.execution.v2.Action.decode(temp_array);
+        this.setState({
+          ...this.state,
+          contents: action_buff,
+          action_array: temp_array,
+          action: temp_action,
+        });
+        this.fetchCommand(temp_action);
+      })
+      .catch(() => {
+        console.error("Error loading bytestream action profile!");
+        this.setState({
+          ...this.state,
+          error: "Error loading action profile. Make sure your cache is correctly configured.",
+        });
+      });
+  }
+
+  displayOutputNodeProps() {
```

Comment:
This whole method can just be rendered below in the render method as:

```jsx
{ this.state.action.outputNodeProperties.length &&
        <div>
          {this.state.action.outputNodeProperties.map((outputNodeProperty) => (
            <div className="output-node">{outputNodeProperty}</div>
          ))}
        </div>
}
{ !this.state.action.outputNodeProperties.length && <div>None found.</div>}
```

or 

```jsx
{ this.state.action.outputNodeProperties.length ?
        <div>
          {this.state.action.outputNodeProperties.map((outputNodeProperty) => (
            <div className="output-node">{outputNodeProperty}</div>
          ))}
        </div> 
: 
<div>None found.</div>
}
```

or if you really want to leave it as a method, you can early exit like 

```ts
displayOutputNodeProps() {
  if  (this.state.action.outputNodeProperties.length == 0) return <div>None found.</div>

  return (<div>
   ...
   ...
   </div>);
}
```

same for the `displayList` function